src/components/Text/Text.test.tsx

```diff
@@ -0,0 +1,27 @@
+import { render, screen } from '@testing-library/react';
+import { Text, TextStyle } from './Text';
+
+describe('Text', () => {
+  it('should render without crashing', () => {
+    const testId = 'test-text';
+    render(<Text data-testid={testId} />);
+    expect(screen.getByTestId(testId)).toBeInTheDocument();
+  });
+
+  it('should display the provided text', () => {
+    const providedText = 'Test text';
+    render(<Text>{providedText}</Text>);
+    expect(screen.getByText(providedText)).toBeInTheDocument();
+  });
+
+  describe('Text styles', () => {
+    Object.entries(TextStyle).forEach(([styleName, styleValue]) => {
+      it(`should match the provided text style ${styleName}`, () => {
```

Comment:
A more precise test name would be
```suggestion
      it(`should apply the appropriate class name for the provided text style ${styleName}`, () => {
```